### PR TITLE
Add lowering of Spread intrinsic

### PR DESCRIPTION
**Pull request by mleair**

### CODE REVIEW

**Comment by psteinfeld** [comment]
All builds, tests, and looks good.  But this intrinsic can accept either an array or a scalar as its first argument.  The test you added in .../flang/test/Lower/intrinsic-procedures.f90 only tests the scalar case, and the test in .../flang/unittests/RuntimeGTest/Transformational.cpp only tests the array case.  It would be nice to test both scalar and array cases in both places.

**Review by psteinfeld — APPROVED** [review]
Aside from wanting additional tests, all looks good.

**Comment by mleair** [comment]
> All builds, tests, and looks good. But this intrinsic can accept either an array or a scalar as its first argument. The test you added in .../flang/test/Lower/intrinsic-procedures.f90 only tests the scalar case, and the test in .../flang/unittests/RuntimeGTest/Transformational.cpp only tests the array case. It would be nice to test both scalar and array cases in both places.

I will add an array case to the regression tests. Our end-to-end tests will also exercise both cases.  I will try to make the change to the unit test in llvm-project.

File: flang/lib/Lower/IntrinsicCall.cpp [hunk 1]
```diff
@@ -2607,6 +2612,38 @@ mlir::Value IntrinsicLibrary::genSpacing(mlir::Type resultType,
...
+IntrinsicLibrary::genSpread(mlir::Type resultType,
+                            llvm::ArrayRef<fir::ExtendedValue> args) {
+
+  assert(args.size() == 3);
+
+  // Handle source argument
+  auto source = builder.createBox(loc, args[0]);
+  fir::BoxValue sourceTmp = source;
+  auto sourceRank = sourceTmp.rank();
+  assert(sourceRank < 15);
```
Comment (schweitzpgi) [inline flang/lib/Lower/IntrinsicCall.cpp:2626]: Similar comment as to Pete's recent PR ... shouldn't the rank be checked in the front end? I wouldn't expect anything to reach lowering with a rank that was invalid.

One advantage of trusting semantics to get this right on its own is that if someone comes along and wants to add a switch to support ranks up to _n_ then lowering can just be agnostic to the issue (for the most part).

**Review by schweitzpgi — APPROVED** [review]
+1

**Pull request merged**